detox/android/detox/src/main/java/com/wix/detox/ReactNativeCompat.java

```diff
@@ -0,0 +1,36 @@
+package com.wix.detox;
+
+import org.joor.Reflect;
+
+import java.util.Map;
+
+public class ReactNativeCompat {
+    static Map<String, Object> VERSION;
+
+    static {
+        try {
+            Class reactNativeVersion = Class.forName("com.facebook.react.modules.systeminfo.ReactNativeVersion");
```

Comment:
This class is present only in RN 50 and above.

Later, the getMinor call could fail on these versions. As we don't differentiate on older versions, I think we could just set it to 49.5.0 in the catch as a default version.

Reply:
Good catch, thanks